src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -580,6 +582,7 @@ private static DataGroup convertStarTableToDataGroup(StarTable table,
 
         // creating DataGroup rows.
         DataGroup dataGroup = new DataGroup(table.getName(), dataTypes);
+        dataGroup.setInitCapacity((int)table.getRowCount());
```

Comment:
Similar to ArrayList, capacity is increased by `oldCapacity * 1.5` when it needs to grow.  At that time, an arrayCopy is performed.  Setting a good initial capacity can avoid unnecessary arrayCopy and unused memory allocation.  It didn't help too much, but every little thing adds up.

The biggest improvement came from using Object[] to represent a row during parsing instead of DataObject, which relies on a HashMap  keyed by column's name to resolve column's data.
It also uses a Map to store column parsed info.  So, 2 Map.get() was needed for each cell value.  In my sample file, a total of 45 million calls were made.  Avoiding these calls were a big gain.